### PR TITLE
Add keybinds for applying queued events and Toggling nametags.

### PR DESCRIPTION
**Pull request by 2PXL**
adds function to toggle nametag with a custom bind (part 2)

this also adds a bind to queue events

### CODE REVIEW

**Comment by Starystars67** [comment]
Hey there, thank you for raising a PR, I notice you mention a part 2, shall we wait for that first before processing your PR?

Thanks

**Comment by 2PXL** [comment]
> Hey there, thank you for raising a PR, I notice you mention a part 2, shall we wait for that first before processing your PR?
> 
> Thanks

I sent that at the same time

**Comment by Starystars67** [comment]
Understood, thank you.
I am just testing the code now, made a small change to fix toggleNametags being null.

**Comment by Starystars67** [comment]
Okay I'm out of time for now, If you have the time would you mind looking into the non working event for `playernametags`, it does not work. I tried changing it to `"onDown": "if MPVehicleGE then MPVehicleGE.hideNicknames() end",` however that did not seem to do it for some reason.

**Comment by 2PXL** [comment]
> Okay I'm out of time for now, If you have the time would you mind looking into the non working event for `playernametags`, it does not work. I tried changing it to `"onDown": "if MPVehicleGE then MPVehicleGE.hideNicknames() end",` however that did not seem to do it for some reason.

I had to press it a couple times before it starts working but ill look into getting it so it works first try

**Comment by 2PXL** [comment]
if it was giving you errors in console I think you also have to add the code I added to "MPVehicleGE.lua"

**Comment by O1LER** [comment]
> if it was giving you errors in console I think you also have to add the code I added to "MPVehicleGE.lua"

But you have to add that commit to this PR. If you go to "Files changed" here at the top right, theres only changes made to the beammp.json

**Comment by O1LER** [comment]
Would be this one b692be8ef88aff4b280d0456cb9b1f1f21222b9b and this 82285cc48000eaa902cdb5a0d5da52ad84bafc8e

But you got them each in its own branch. You can make a branch for a certain change/feature that you are working on, and this branch can have multiple commits in it.
Or you can work directly on your fork of the development branch.
Dont know which method ispreferred by beamMP

**Comment by 2PXL** [comment]
> Would be this one [b692be8](https://github.com/BeamMP/BeamMP/commit/b692be8ef88aff4b280d0456cb9b1f1f21222b9b) and this [82285cc](https://github.com/BeamMP/BeamMP/commit/82285cc48000eaa902cdb5a0d5da52ad84bafc8e)
> 
> But you got them each in its own branch. You can make a branch for a certain change/feature that you are working on, and this branch can have multiple commits in it. Or you can work directly on your fork of the development branch. Dont know which method ispreferred by beamMP

can I make a branch now with my changes? or should I remember that for next time?

**Comment by O1LER** [comment]
I think you can move those commits over, either directly into this PR, or into one of the branches you created.
Or, you merge each of your branches into your development branch, then open up a new PR here that merges your development branch into the beamMP one
But i dont know which of those options if the best

**Comment by Starystars67** [comment]
> Would be this one [b692be8](https://github.com/BeamMP/BeamMP/commit/b692be8ef88aff4b280d0456cb9b1f1f21222b9b) and this [82285cc](https://github.com/BeamMP/BeamMP/commit/82285cc48000eaa902cdb5a0d5da52ad84bafc8e)
> 
> But you got them each in its own branch. You can make a branch for a certain change/feature that you are working on, and this branch can have multiple commits in it. Or you can work directly on your fork of the development branch. Dont know which method ispreferred by beamMP

The preferred is to use a new "feature" branch under your own fork for new features, and a similar convention for fixes/issues and the likes. That said working on the development branch is also okay since it will be brought in via PR.

**Comment by Starystars67** [comment]
> > Would be this one [b692be8](https://github.com/BeamMP/BeamMP/commit/b692be8ef88aff4b280d0456cb9b1f1f21222b9b) and this [82285cc](https://github.com/BeamMP/BeamMP/commit/82285cc48000eaa902cdb5a0d5da52ad84bafc8e)
> > But you got them each in its own branch. You can make a branch for a certain change/feature that you are working on, and this branch can have multiple commits in it. Or you can work directly on your fork of the development branch. Dont know which method ispreferred by beamMP
> 
> can I make a branch now with my changes? or should I remember that for next time?

You did already make a branch, it is called 2PXL:2PXL-patch-1-1. Please do follow O1LER's advice and have all of your work relating to this in one branch so that it will all be a part of the same PR.

**Comment by 2PXL** [comment]
> > > Would be this one [b692be8](https://github.com/BeamMP/BeamMP/commit/b692be8ef88aff4b280d0456cb9b1f1f21222b9b) and this [82285cc](https://github.com/BeamMP/BeamMP/commit/82285cc48000eaa902cdb5a0d5da52ad84bafc8e)
> > > But you got them each in its own branch. You can make a branch for a certain change/feature that you are working on, and this branch can have multiple commits in it. Or you can work directly on your fork of the development branch. Dont know which method ispreferred by beamMP
> > 
> > 
> > can I make a branch now with my changes? or should I remember that for next time?
> 
> You did already make a branch, it is called 2PXL:2PXL-patch-1-1. Please do follow O1LER's advice and have all of your work relating to this in one branch so that it will all be a part of the same PR.

done, hope that was the correct way.

**Comment by Starystars67** [comment]
> done, hope that was the correct way.

Yep, spot on!

File: lua/ge/extensions/MPVehicleGE.lua [hunk 1]
```diff
@@ -252,6 +252,19 @@ function getGameVehicleID(serverVehicleID)
 	end
 end
 
+-- Global variable to keep track of the current state
```
Comment (Starystars67) [inline lua/ge/extensions/MPVehicleGE.lua]: Please can you comment this according to the format we are using which is ldoc:
https://github.com/lunarmodules/ldoc

File: lua/ge/extensions/MPVehicleGE.lua [hunk 2]
```diff
@@ -252,6 +252,19 @@ function getGameVehicleID(serverVehicleID)
...
+local hideNicknames = false
```
Comment (Starystars67) [inline lua/ge/extensions/MPVehicleGE.lua]: Please move this variable up to be with the rest of our defined variables.

**Review by Starystars67 — CHANGES_REQUESTED** [review]
Once the two changes are made and an in-game test is done I am happy with this to be merged.

Thank you

**Comment by 2PXL** [comment]
should be good now if I understood it correctly.

**Review by Starystars67 — APPROVED** [review]

**Comment by Starystars67** [comment]
Hello, I made a couple of small changes, nothing major but all good now.
Thank you for your contribution to the project.

Is there anything else you would like to add before we go ahead and merge this?

**Review by Starystars67 — APPROVED** [review]

**Comment by 2PXL** [comment]
> Hello, I made a couple of small changes, nothing major but all good now. Thank you for your contribution to the project.
> 
> Is there anything else you would like to add before we go ahead and merge this?

no that is all, thank you.

**Comment by Starystars67** [comment]
Awesome, Thank you for your contribution towards this!
If you @ me (Titch) in the BeamMP Discord and DM me on our forum I will drop you the contributor roles so you can join the rest of the team.

**Pull request merged**